File: src/api/getEmployeesAndChairs.ts
import axios from 'axios';
import { KeyValue } from '.';


export interface EmployeesAndChairs {
    __type: string;
    chairs: KeyValue[];
    employees: KeyValue[];
    studyTypes?: KeyValue[];
}
export const getEmployeesAndChairs = async (facultyId: string): Promise<EmployeesAndChairs> => {

    const response = await axios('https://vnz.osvita.net/BetaSchedule.asmx/GetEmployeeChairs', {
        params: {
            callback: '',
            aVuzID: 11613,
            aFacultyID: `"${facultyId}"`,
            aGiveStudyTimes: 'false',
        },
    });

    const data = response.data.d;
    if (!data) throw new Error('Failed to get employees');
    
    return data;
}

File: api/get_employees_and_chairs.go
package api

import (
	"encoding/json"
	"io"
	"net/http"
)

type EmployeesAndChairs struct {
	Employees []KeyValue `json:"employees"`
	Chairs    []KeyValue `json:"chairs"`
}

type getEmployeesAndChairsResponse struct {
	D EmployeesAndChairs `json:"d"`
}

func GetEmployeesAndChairs(faculty string) (*EmployeesAndChairs, error) {
	req, _ := http.NewRequest("GET", "https://vnz.osvita.net/WidgetSchedule.asmx/GetEmployeeChairs", nil)

	q := req.URL.Query()
	q.Add("callback", "")
	q.Add("aVuzID", "11613")
	q.Add("aFacultyID", "\""+faculty+"\"")
	q.Add("aGiveStudyTimes", "false")
	req.URL.RawQuery = q.Encode()

	res, err := http.DefaultClient.Do(req)
	if err != nil {
		return nil, err
	}

	body, err := io.ReadAll(res.Body)
	if err != nil {
		return nil, err
	}

	var result getEmployeesAndChairsResponse
	err = json.Unmarshal(body, &result)
	if err != nil {
		return nil, err
	}

	return &result.D, nil
}
